src/plugins/expressions/common/expression_types/filter.ts

```diff
@@ -34,9 +35,35 @@ export interface Filter {
   query?: string | null;
 }
 
+const migrator = {
+  isUnsupported: (p: any) => !p.version || p.version < 2,
+  isLegacy: (p: any) => p.version && p.version < 3,
+  migrate: (legacyFilter: any) => ({ ...legacyFilter, version: 3 }),
+};
```

Comment:
This is some throwaway code to run a fake migration which simply increments the version number. If version 1 is provided, we consider this to be no longer supported and throw an error. If version 2 is provided, we provide the ability to migrate it up to the latest (version 3).

How these migrations are structured and standardized is [a separate topic](https://github.com/elastic/kibana/issues/56166); this is only an example.